lib/Basics/ReadWriteLock.cpp

```diff
@@ -50,6 +50,33 @@ void ReadWriteLock::writeLock() {
   }
 }
 
+/// @brief lock for writes with microsecond timeout
+bool ReadWriteLock::writeLock(std::chrono::microseconds timeoutx) {
```

Comment:
why is the variable named `timeoutx`? what does the `x` stand for?

Reply:
x stands for, what the hell is keeping this function from compiling, is timeout somehow a keyword?  It was not, but I forgot to remove x.  That and all other changes are about to check in.